### PR TITLE
Add modules that document the initial data pipeline and workflow

### PR DESCRIPTION
**Pull request by nordzilla**
All naming conventions, types, and descriptions are heavily subject to change. The intent of this pull request is to provide a concrete structure for a starting point to begin exploring the different components that will be required for this project.

### CODE REVIEW

File: src/doc_gen/mod.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//! Handles generating documentation comments from a `DocTree`.
+
+/// The `DocGenerator` is a type that consumes `DocTree` nodes
+/// and produces a flatened list of formatted documentation comments,
```
Comment (connorkuehl) [inline src/doc_gen/mod.rs]: typo: 'flatened' -> 'flattened'

File: src/check/mod.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//! The main entry point for the `doc-gen --check` command option.
+//! Handles validating that existing documentation in a rust project
+//! conforms to the specifications set by `DocConfig`.
+//!
+//! Should also be capable of determining if documtnation is out of date
```
Comment (connorkuehl) [inline src/check/mod.rs]: typo: 'documtnation' -> 'documentation'

File: src/ast/mod.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+//! Handles generating an abstract syntax tree of a Rust project.
+
+/// An AST that represents a rust project in a given directory.
+/// The intent is to reuse as much as possible from the existing Rust
+/// parser from rustc. We may aggregate additional information or remove
+/// information from indiviual tree nodes depending on our needs.
```
Comment (connorkuehl) [inline src/ast/mod.rs]: typo: 'indiviual' -> 'individual'

**Review by connorkuehl — CHANGES_REQUESTED** [review]
LGTM with typos fixed :+1:

**Comment by nordzilla** [comment]
@connorkuehl Thanks for catching those typos. PTAL.

File: src/doc_config/mod.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//! Handles the configuration of how documentation comments should be
+//! generated by `doc-gen`.
+//!
+//! Documentation will likely be written in a TOML syntax and deserialized/validated
```
Comment (squidboylan) [inline src/doc_config/mod.rs]: should this say documentation config or template rather than just "documentation"?

**Review by connorkuehl — APPROVED** [review]

File: src/doc_config/mod.rs (hunk 4 above)
Comment (nordzilla) [reply src/doc_config/mod.rs]: Definitely should say config files. Thanks!

**Pull request merged**